Modular monolith

BBSRKhelp opened the pull request.

BBSRKhelp committed:
Starting modular-monolith. Create module species

BBSRKhelp committed:
Modify the species model

Dependencies and using have tightened up, moved several directories

BBSRKhelp committed:
Put the main modules in their places and set up connection

The main modules are in place.
The links in .sln and .csproj have been changed to correct ones.
The project is not launching yet.
Communication with Docker is not configured.
The schemas in the database are not separated.
The tests cause errors and it has not yet been decided where to store them.

BBSRKhelp committed:
Fix class Program and Tests

Fixed the logs in the Program class.
Set up the tests to make them work

BBSRKhelp committed:
The modular monolith is ready for the first delivery.

The application can be launched and it will work.
Moved requests to the presentation layer from the contracts layer.
Returned and updated Apple Migration

BBSRKhelp commented:
1. Не знаю куда деть папку Tests. Нужно ли ее раскидывать между Species и Volunteer? И стоит ли назвать tests вместо заглавной?
2. Оставил requests в presentation, чтобы не париться с реализацией маппинга в command. Или все же лучше перекинуть в Contracts? 
3. В данный момент чтобы работать с приложением создаю две миграции в Species.Infrast.., и в Volunteer.Infras... . ApplyMigration перенес в Web, изначально хотел в Shared.Framework, но оттуда, я не смог бы контексты базы данных подтянуть. Вот не знаю хорошее ли это место для ApplyMigration в Web. Метод реализацию положил там в папку Extensions, не лучше ли положить ее в Middlewares, хоть и не является, но при каждом билде проекта вызывается, то есть чем то, да похоже на мидлвеер.
4. В каждом модуле все DI (Из Application и Infrastructure) передаю в Presentation и далее уже из Presentation всех модулей в Program. Выглядит прикольно, распределенно довольно. Но из за этого в Presentation идет зависимость от Infrastrucuture не плохо ли это?
5. В Core установил Npgsql.EFCore.PostreSql. В модуле Volunteer в Application использую метод ILIKE(по типу Contains) при получении всех волонтеров, этот метод доступен только в либе Npgsql. Также SqlConnectionFactory у меня лежит в Core, а он также использует Npgsql. Не смертельно?

n-cherashev commented on PetFamily.Backend/src/File/PetFamily.File.Presentation/PetFamily.File.Presentation.csproj:
```diff
@@ -0,0 +1,15 @@
+
+    <ItemGroup>
+      <ProjectReference Include="..\PetFamily.File.Application\PetFamily.File.Application.csproj" />
+      <ProjectReference Include="..\PetFamily.File.Infrastructure\PetFamily.File.Infrastructure.csproj" />
```
У нас Presentation не должен ссылаться на Infrastructure

![image](https://github.com/user-attachments/assets/5b11bb96-5bad-4316-b903-39381a3c7e91)

n-cherashev commented on PetFamily.Backend/src/Species/PetFamily.Species.Presentation/PetFamily.Species.Presentation.csproj:
```diff
@@ -0,0 +1,16 @@
+      <ProjectReference Include="..\..\Shared\PetFamily.Framework\PetFamily.Framework.csproj" />
+      <ProjectReference Include="..\PetFamily.Species.Application\PetFamily.Species.Application.csproj" />
+      <ProjectReference Include="..\PetFamily.Species.Contracts\PetFamily.Species.Contracts.csproj" />
+      <ProjectReference Include="..\PetFamily.Species.Infrastructure\PetFamily.Species.Infrastructure.csproj" />
```
У нас Presentation не должен ссылаться на Infrastructure

n-cherashev commented on PetFamily.Backend/src/Volunteer/PetFamily.Volunteer.Presentation/PetFamily.Volunteer.Presentation.csproj:
```diff
@@ -0,0 +1,16 @@
+      <ProjectReference Include="..\..\Shared\PetFamily.Framework\PetFamily.Framework.csproj" />
+      <ProjectReference Include="..\PetFamily.Volunteer.Application\PetFamily.Volunteer.Application.csproj" />
+      <ProjectReference Include="..\PetFamily.Volunteer.Contracts\PetFamily.Volunteer.Contracts.csproj" />
+      <ProjectReference Include="..\PetFamily.Volunteer.Infrastructure\PetFamily.Volunteer.Infrastructure.csproj" />
```
У нас Presentation не должен ссылаться на Infrastructure

n-cherashev reviewed (changes requested):
Ответы на твои вопросы:
1. Папку Tests можно оставить общей и там называть проекты например PetFamily.Files.Application.UnitTests, PetFamily.Volunteers.Application.UnitTests  и тд
2. Лучше сделать в Contracts, а в Presentation просто создавать через оператор new нужные комманды
3. Лучше в Web оставь
4. Лучше в Web вызывай просто extention метод каждого слоя каждого модуля, не нужно делать ссылку на Infrastrucuture
5. Нормально, так можно делать

BBSRKhelp committed:
Refactoring

Changed the structure of the tests.
All tests work out positively.
Removed the infrastructure dependency from presentation.
Created extensions methods for mapping from request to command
